File: Design-Patterns-Using-Kotlin/Basics OOPS/Polymorphism.kt
class Adder {
    var value: Int = 0
        private set

    // Overloaded method for incrementing by a specific value
    fun increment(a: Int) {
        this.value += a
    }

    // Overloaded method for incrementing by 1
    fun increment() {
        this.value += 1
    }
}

fun main() {
    val a = Adder()
    a.increment(10)
    a.increment()
    println(a.value)
}

/*
11
*/

File: Design-Patterns-Using-Swift/Basics OOPS/Polymorphism.swift
class Adder {
    private var sum: Int

    init() {
        self.sum = 0
    }

    // Overloaded method for incrementing by a specific value
    func increment(_ a: Int) {
        self.sum += a
    }

    // Overloaded method for incrementing by 1
    func increment() {
        self.sum += 1
    }

    func getValue() -> Int {
        return self.sum
    }
}

let a = Adder()
a.increment(10)
a.increment()
print(a.getValue())
